modules/core/src/lib/layer.js

```diff
@@ -741,6 +741,11 @@ export default class Layer extends Component {
       if (flags[key] && !changeFlags[key]) {
         changeFlags[key] = flags[key];
         debug(TRACE_CHANGE_FLAG, this, key, flags);
+      } else if (flags[key] && key === 'dataChanged' && Array.isArray(changeFlags[key])) {
```

Comment:
Nit: I see we are having one changeFlag of a different type. 

Understand why but it is a little messy. 

For clarity around this, even though it would require some duplicated code, I would add an outer switch on the `key`.

```js
for (const key in changeFlags) {
  switch (key) {
     case 'dataChanged':
       // comment reminding reader what type of dataChanged is
       ..
     default: 
       if (flags[key] && !changeFlags[key]) {
         changeFlags[key] = flags[key];
         debug(TRACE_CHANGE_FLAG, this, key, flags);
       }
```